File: R/combine_padding.R
#' Add / remove padding
#'
#' @param session shiny session.
#' @param inputId Input's id to add / remove.
#' @param type 'regular' or 'combine'.
#'
#' @noRd
combine_padding <- function(session, inputId, type = "regular") {
  session$sendCustomMessage(
    type = "method_combine_padding",
    message = list(inputId = inputId, type = type)
  )
}

File: inst/js/combine_padding.js
// Disable / enable a button
Shiny.addCustomMessageHandler('method_combine_padding', function(data) {
  if (data.type == 'combine') {
    $('#' + data.inputId).addClass('combine_padding');
  }
  if (data.type == 'regular') {
    $('#' + data.inputId).removeClass('combine_padding');
  }
});
